### ISSUE TITLE
Improve current way name label

### ISSUE THREAD

**Issue by bsudekum** [mapbox/mapbox-navigation-ios#193]
![image](https://cloud.githubusercontent.com/assets/1058624/25598100/d9051bbc-2e87-11e7-85f7-273c6f5c2651.png)

This should be US-101 instead. If a way name includes a `ref`, we should use this instead.

/cc @1ec5

**Comment by 1ec5** [comment]
It should include both the shield and the name, since we don't have the data to know which is more relevant. (Think county and state route numbers on surface streets.)

**Comment by 1ec5** [comment]
We do know the current road’s classification (especially after #297). In general, we can assume that the shield is more important for freeways. However, I still think we should show both if possible, just in case.

**Comment by ericrwolfe** [comment]
This should be improved with the changes in #291.

**Comment by 1ec5** [comment]
Is this issue about the label at the bottom of the map (the “current way name label”) or the upcoming way name label in the turn banner? The former has not been improved since its original implementation.

**Comment by ericrwolfe** [comment]
@1ec5 oh yes, you are correct. The current way name label is indeed the small label at the bottom of the map. 

The logic used for fixing #291 could also be used to determine the proper current way name label.

**Comment by ericrwolfe** [comment]
> we can assume that the shield is more important for freeways

A shield is probably too small for the way name label here, especially since shields already appear in the map along the route line, so using the textual `US-101` would be fine to start.

**Comment by 1ec5** [comment]
> using the textual `US-101` would be fine to start

The street name currently comes from Mapbox Streets vector tiles because of how OSRM collapses name change instructions. U.S. Route 101 would be represented in the vector tiles as `shield: us-highway` `ref: 101`. We’ll need to map each of [these `shield` values](https://www.mapbox.com/vector-tiles/mapbox-streets-v7/#road_label) to a recognizable prefix.

**Comment by bsudekum** [comment]
Closing in favor of https://github.com/mapbox/mapbox-navigation-ios/issues/590
